src/irr/asset/CGLSLVirtualTexturingBuiltinIncludeLoader.h

```diff
@@ -0,0 +1,368 @@
+#ifndef _IRR_C_GLSL_VIRTUAL_TEXTURING_BUILTIN_INCLUDE_LOADER_H_INCLUDED__
+#define _IRR_C_GLSL_VIRTUAL_TEXTURING_BUILTIN_INCLUDE_LOADER_H_INCLUDED__
+
+#include "irr/asset/CGraphicsPipelineLoaderMTL.h"
+
+namespace irr {
+namespace asset
+{
+
+class CGLSLVirtualTexturingBuiltinIncludeLoader : public IBuiltinIncludeLoader
+{
+public:
+    const char* getVirtualDirectoryName() const override { return "glsl/virtual_texturing/"; }
+
+private:
+	static core::vector<std::string> parseArgumentsFromPath(const std::string& _path)
+	{
+		core::vector<std::string> args;
+
+		std::stringstream ss{ _path };
+		std::string arg;
+		while (std::getline(ss, arg, '/'))
+			args.push_back(std::move(arg));
+
+		return args;
+	}
+	static std::string getDescriptors(const std::string& _path)
+	{
+		auto args = parseArgumentsFromPath(_path.substr(_path.find_first_of('/') + 1, _path.npos));
+		constexpr uint32_t
+			set_ix		= 0u,
+			pgt_bnd_ix	= 1u,
+			f_bnd_ix	= 2u,
+			i_bnd_ix	= 3u,
+			u_bnd_ix	= 4u,
+			f_cnt_ix	= 5u,
+			i_cnt_ix	= 6u,
+			u_cnt_ix	= 7u;
+
+		return 
+"\n#ifndef _IRR_VT_DESCRIPTOR_SET"
+"\n#define _IRR_VT_DESCRIPTOR_SET " + args[set_ix] +
+"\n#endif" +
+"\n#ifndef _IRR_VT_PAGE_TABLE_BINDING"
+"\n#define _IRR_VT_PAGE_TABLE_BINDING " + args[pgt_bnd_ix] +
+"\n#endif" +
+"\n#ifndef _IRR_VT_FLOAT_VIEWS"
+"\n#define _IRR_VT_FLOAT_VIEWS_BINDING " + args[f_bnd_ix] +
+"\n#define _IRR_VT_FLOAT_VIEWS_COUNT " + args[f_cnt_ix] +
+"\n#endif" +
+"\n#ifndef _IRR_VT_INT_VIEWS"
+"\n#define _IRR_VT_INT_VIEWS_BINDING " + args[i_bnd_ix] +
+"\n#define _IRR_VT_INT_VIEWS_COUNT " + args[i_cnt_ix] +
+"\n#endif" +
+"\n#ifndef _IRR_VT_UINT_VIEWS"
+"\n#define _IRR_VT_UINT_VIEWS_BINDING " + args[u_bnd_ix] +
+"\n#define _IRR_VT_UINT_VIEWS_COUNT " + args[u_cnt_ix] +
+"\n#endif" +
```

Comment:
you could just provide sane defaults here, and let user override

or if providing defaults don't really make sense, you could check if user has defined them and if not emit an `#error`